.github/workflows/publish-test-results.yml

```diff
@@ -82,10 +85,15 @@ jobs:
           max_retries=3
           sleep_time=5  # seconds
           status=1  # any non-zero value
+          auth_header=""
+          if [ -n "${{ secrets.AUTO_SCORE_TOKEN }}" ]; then
+           auth_header="Authorization: Bearer ${{ secrets.AUTO_SCORE_TOKEN }}"
+          fi
           while [[ "$status" -ne 0 && "$retry" -lt "$max_retries" ]]; do
             echo "Attempt: $((retry+1))"
             curl -s -o /dev/null -w "%{http_code}" \
               -X POST \
+              -H "$auth_header" \
```

Comment:
only added when `secrets. AUTO_SCORE_TOKEN` is provided, curl supports the default flow `-H "" `